src/box/vy_run.c

```diff
@@ -969,44 +998,111 @@ static int
 vy_page_read_cb(struct cbus_call_msg *base)
 {
 	struct vy_page_read_task *task = (struct vy_page_read_task *)base;
-	ZSTD_DStream *zdctx = vy_env_get_zdctx(task->run->env);
+	ZSTD_DStream *zdctx = vy_env_get_zdctx(task->run->run_env);
 	if (zdctx == NULL)
 		return -1;
 	if (vy_page_read(task->page, task->page_info, task->run, zdctx) != 0)
 		return -1;
 	if (task->key.stmt != NULL) {
 		task->pos_in_page = vy_page_find_key(task->page, task->key,
-						     task->cmp_def, task->format,
+						     task->cmp_def,
+						     task->format,
 						     task->iterator_type,
 						     &task->equal_found);
 	}
 	return 0;
 }
 
 /**
- * Increase memory-stored page's reference counter.
+ * Delete fixed percentage of pages contained in page cache's LRU list -
+ * starting from the least recently used one and further towards the head
+ * of the list. At least one page is deleted in any case, except in the case
+ * of empty list.
+ */
+static inline void										//temp/ maybe should add error codes (what errors?..)
+vy_page_cache_lru_cutback(struct vy_run *run, uint8_t del_percent)
+{
+	assert(del_percent <= 100);
+	uint32_t npages_to_del = run->page_cache_lru_size -
+			(run->page_cache_lru_size *
+			(100 - del_percent) / 100);					//temp/ always more than enough
+
+	for (uint32_t i = 0; i < npages_to_del; ++i) {
+		assert(!rlist_empty(&run->page_cache_lru));
+		struct vy_page_info *tail_item =
+				rlist_shift_tail_entry(&run->page_cache_lru,
+						       struct vy_page_info,
+						       in_lru);				//temp/ unnecessary to unlink node and link next tail
+		struct vy_page *tail_page = tail_item->page;
+		assert(tail_page->ri_refs == 0);
+		run->run_env->pc_env.mem_used -= (tail_item->unpacked_size +
+				sizeof(struct vy_page) + tail_item->row_count *
+				sizeof(*tail_page->row_index));
+		vy_page_delete(tail_page);
+		tail_item->page = NULL;
+	}
+	run->page_cache_lru_size -= npages_to_del;
+}
+
+/**
+ * Delete fixed percentage of cached (stored but not used) pages in each run's
+ * page cache.
+ * Pages are deleted in every page cache starting from the least recently used
+ * page and further towards the head of LRU list - until requested
+ * percentage of pages in current page cache is deleted.
+ */
+static void										//temp/ maybe should add error codes (what errors?..)
+vy_page_cache_mem_cutback(struct vy_run_env *run_env, uint8_t del_percent)
```

Comment:
Let's use percentage instead of ratio.

Reply:
Guess you meant "ratio instead of percentage". Anyway, we already abandoned the percentage-based cutbacks and switched to a classic "rotating LRU list" scheme